sopel/modules/reddit.py

```diff
@@ -328,3 +384,38 @@ def get_channel_spoiler_free(bot, trigger):
         bot.say('%s is flagged as spoiler-free' % channel)
     else:
         bot.say('%s is flagged as spoilers-allowed' % channel)
+
+
+@rule(r'^(?P<prefix>r|u)\/(?P<id>[a-zA-Z-_]+)')
```

Comment:
Digits are allowed in names on reddit. Also, forward slash needn't be escaped in Python regex.

```suggestion
@rule(r'^(?P<prefix>r|u)/(?P<id>[a-zA-Z0-9-_]+)')
```